vochain/types.go

```diff
@@ -59,13 +64,74 @@ type QueryData struct {
 	ProcessType string `json:"type,omitempty"`
 }
 
+// ________________________ TRANSACTION COSTS __________________________
+// TransactionCosts describes how much each operation should cost
+type TransactionCosts struct {
+	SetProcess            uint64 `json:"Tx_SetProcess" state:"c_setProcess"`
+	RegisterKey           uint64 `json:"Tx_RegisterKey" state:"c_registerKey"`
+	NewProcess            uint64 `json:"Tx_NewProcess" state:"c_newProcess"`
+	SendTokens            uint64 `json:"Tx_SendTokens" state:"c_sendTokens"`
+	SetAccountInfo        uint64 `json:"Tx_SetAccountInfo" state:"c_setAccountInfo"`
+	AddDelegateForAccount uint64 `json:"Tx_AddDelegateForAccount" state:"c_addDelegateForAccount"`
+	DelDelegateForAccount uint64 `json:"Tx_DelDelegateForAccount" state:"c_delDelegateForAccount"`
+	CollectFaucet         uint64 `json:"Tx_CollectFaucet" state:"c_collectFaucet"`
+}
+
+// StructAsBytes returns the contents of TransactionCosts as a map. Its purpose
+// is to keep knowledge of TransactionCosts' fields within itself, so the
+// function using it only needs to iterate over the key-values.
+func (t *TransactionCosts) StructAsBytes() (b map[string][]byte, err error) {
+	b = make(map[string][]byte)
+
+	tType := reflect.TypeOf(*t)
+	tValue := reflect.ValueOf(*t)
+	for i := 0; i < tType.NumField(); i++ {
+		key := tType.Field(i).Tag.Get("state")
+		value := tValue.Field(i).Uint()
+
+		valueBytes, err := util.Uint64ToBytes(value)
+		if err != nil {
+			return nil, err
+		}
+
+		b[key] = valueBytes
+	}
+	return
+}
+
+// TransactionCostsFieldFromStateKey transforms "c_setProcess" to "SetProcess" for all of
+// TransactionCosts' fields
+func TransactionCostsFieldFromStateKey(key string) (string, error) {
+	if key[0:2] != costPrefix {
+		return "", fmt.Errorf("state keys must start with '%s', got %s", costPrefix, key)
+	}
+	name := strings.TrimLeft(key, costPrefix)
+
+	// strings.Title will misbehave when there are punctuation marks in the
+	// string. To clean the input up, we ensure there are only alphabetical
+	// characters left in the string
+	isAlpha := regexp.MustCompile(`^[A-Za-z]+$`).MatchString
+	if !isAlpha(name) {
+		return "", fmt.Errorf("%s needs to be alphabetical only", name)
+	}
+	Name := strings.Title(name)
```

Comment:
Not go friendly, please don't use capital letter for function scope variables.

Reply:
Aww i thought it was an elegant way of expressing what that variable was supposed to contain